### PR TITLE
MAINT: Remove environment as an argument to benchmark source.

### PR DESCRIPTION
**Pull request by ehebert**
To allow the BenchmarkSource class to be more easily used in contexts other than
a TradingAlgorithm, remove the TradingEnvironment as an argument to the
benchmark source.

Instead:
- Pass a benchmark Asset, instead of a bencmark sid; so that the asset_finder
does not need to be passed to the benchmark source.
- Pass the pre-calculated benchmark_returns instead of an env,
which contains the benchmark_returns; a consumer can let the benchmark_returns
stay as the default of `None` when using an asset.

We may want to further refactor and make two different classes, instead of
relying on a combination of existence/non-existence of benchmark_asset and
benchmark_returns. That refactoring should be easier to do with this change.

### CODE REVIEW

File: zipline/sources/benchmark_source.py [hunk 1]
```diff
@@ -46,11 +48,10 @@ def __init__(self, benchmark_sid, env, trading_calendar, sessions,
                     self.sessions,
                     self.data_portal
                 )
-        else:
+        elif benchmark_returns is not None:
             # get benchmark info from trading environment, which defaults to
             # downloading data from Yahoo.
-            daily_series = \
-                env.benchmark_returns[sessions[0]:sessions[-1]]
+            daily_series = benchmark_returns[sessions[0]:sessions[-1]]
```
Comment (jbredeche) [inline zipline/sources/benchmark_source.py]: the comment just above this line needs updating now

Comment (ehebert) [inline zipline/sources/benchmark_source.py]: Good catch. I'll move that to where the returns are retrieved in `algorithm.py`.

File: zipline/sources/benchmark_source.py [hunk 2]
```diff
@@ -77,27 +79,27 @@ def _validate_benchmark(self, benchmark_asset):
         # error suggesting that the user pick a different asset to use
         # as benchmark.
         stock_dividends = \
-            self.data_portal.get_stock_dividends(self.benchmark_sid,
+            self.data_portal.get_stock_dividends(self.benchmark_asset,
```
Comment (srklonaris) [inline zipline/sources/benchmark_source.py]: Does this method already require a sid or an asset?

Comment (ehebert) [inline zipline/sources/benchmark_source.py]: `get_stock_dividends` accepts either `Asset` or an `int`. (It calls `int` on the `sid` value before creating the sql query.)
The docstring should probably be updated on that method to reflect that either is accepted; though in general we have aimed to pass `Asset`s to `DataPortal` methods.

Comment (srklonaris) [inline zipline/sources/benchmark_source.py]: okay, makes sense

**Comment by srklonaris** [comment]
lgtm!  Thanks for doing this, its really helpful for me when needing to build a `BenchmarkSource` in qids

File: zipline/sources/benchmark_source.py [hunk 3]
```diff
@@ -68,6 +67,9 @@ def __init__(self, benchmark_sid, env, trading_calendar, sessions,
                 self._precalculated_series = minute_series
             else:
                 self._precalculated_series = daily_series
+        else:
+            raise Exception("Must provide either benchmark_asset or "
```
Comment (richafrank) [inline zipline/sources/benchmark_source.py]: What do you think about `ValueError` or something else more specific?

**Pull request merged**